File: backend/process-excel.py
import pandas as pd
import sqlite3
import os
from pathlib import Path

# Read the Excel file
excel_path = r"C:\Users\MostafaMegahedTAQAHo\Downloads\10-11-2024 FAINAL.xlsx"
print(f"Reading Excel file: {excel_path}")

try:
    # Read all sheets to see what's available
    xl_file = pd.ExcelFile(excel_path)
    print(f"Available sheets: {xl_file.sheet_names}")
    
    # Read the first sheet (or specify sheet name if known)
    df = pd.read_excel(excel_path, sheet_name=0)
    
    print(f"Columns found: {list(df.columns)}")
    print(f"Total rows: {len(df)}")
    print("\nFirst few rows:")
    print(df.head())
    
    # Save to CSV for easier inspection
    csv_path = r"C:\Users\MostafaMegahedTAQAHo\OneDrive - TAQA Arabia\Mustafa\dragon lap\backend\laptops_data.csv"
    df.to_csv(csv_path, index=False)
    print(f"\nData saved to CSV: {csv_path}")
    
except Exception as e:
    print(f"Error reading Excel file: {e}")
    print("Please make sure the file exists and is accessible.")

File: backend/process-excel.js
const XLSX = require('xlsx');
const fs = require('fs');
const path = require('path');

// Read the Excel file
const excelPath = "C:\\Users\\MostafaMegahedTAQAHo\\Downloads\\10-11-2024 FAINAL.xlsx";

console.log(`Reading Excel file: ${excelPath}`);

try {
    // Check if file exists
    if (!fs.existsSync(excelPath)) {
        console.error("Excel file not found!");
        process.exit(1);
    }

    // Read the workbook
    const workbook = XLSX.readFile(excelPath);
    
    // Get sheet names
    console.log("Available sheets:", workbook.SheetNames);
    
    // Read the first sheet
    const firstSheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[firstSheetName];
    
    // Convert to JSON
    const data = XLSX.utils.sheet_to_json(worksheet);
    
    console.log(`Total rows: ${data.length}`);
    console.log("Columns found:", Object.keys(data[0] || {}));
    console.log("\nFirst few rows:");
    console.log(data.slice(0, 3));
    
    // Save to JSON file for easier processing
    const jsonPath = path.join(__dirname, 'laptops_data.json');
    fs.writeFileSync(jsonPath, JSON.stringify(data, null, 2));
    console.log(`\nData saved to JSON: ${jsonPath}`);
    
} catch (error) {
    console.error("Error reading Excel file:", error.message);
}